File: src/main/java/legacy/kms/chapter02/discount/NoneDiscountPolicy.java
package legacy.kms.chapter02.discount;

import legacy.kms.chapter02.Money;
import legacy.kms.chapter02.Screening;

public class NoneDiscountPolicy implements DiscountPolicy {

    @Override
    public Money calculateDiscountAmount(Screening screening) {
        return Money.ZERO;
    }
}

File: src/main/kotlin/ezhoon/chapter02/policy/NoneDiscountPolicy.kt
package ezhoon.chapter02.policy

import ezhoon.chapter02.Money
import ezhoon.chapter02.Screening
import ezhoon.chapter02.condition.DiscountCondition

class NoneDiscountPolicy(
    override val conditions: List<DiscountCondition>
) : DiscountPolicy {

    override fun calculateDiscountAmount(screening: Screening): Money {
        return getDiscountAmount(screening)
    }

    override fun getDiscountAmount(screening: Screening): Money {
        return Money.ZERO
    }
}